### DISCUSSION TITLE
Failed to retrieve map from vacuum

### DISCUSSION THREAD

**Discussion by lkw11** [PiotrMachowski/Home-Assistant-custom-components-Xiaomi-Cloud-Map-Extractor#298] [Q&A]
Hi guys, 

Sorry if I asked another stupid question. I knew this has been asked few times and I read all other post and still could not fix this issue

I have used this extractor and able to control the vacuum without problem but the only issue I have is the map always return to " Failed to retrieve map from vacuum " status after few hours working. 

I need to open my MiHome app on my phone and wait for the map to load up and home assistant will then update the map and it will work again. 

Is there anything I need to setup in MiHome app or HA to keep the map alive or automatic update everything it failed to retrieve map ?

Many thanks

**Comment by PiotrMachowski** [comment]
Try restarting your vacuum

**Reply by lkw11** [reply]
Thanks for the reply.
Do you mean I power off the vacuum and power on again or restarting the HA ?

**Reply by PiotrMachowski** [reply]
Power off the vacuum and turn it back on

**Reply by lkw11** [reply]
Thanks I will try this.

**Reply by lkw11** [reply]
Hi Piotr,

Just an update, I restarted the vacuum but same issue, it was working and couple hours later it show Failed to retrieve map from vacuum.
After I start the MiHome app and the vacuum will update the map and everything back to working, but it only last for few hours.

**Reply by PiotrMachowski** [reply]
What vacuum model do you have?

**Reply by lkw11** [reply]
I have the Roborock G10.

**Comment by radinsky** [comment]
same here. 
the only workaround (beside the vacuum restart) is to connect to vacuum within MI app and then it immediately starts working as well the map extractor

**Reply by adjbuzz** [reply]
For me, this issue was caused because the IP for my vacuum changed which made the IP listed for the "host:" in the config.yaml incorrect. Once I updated that, everything loaded like normal. Also reserved that IP in my router to prevent that from happening going forward. Hope this helps!

**Reply by lkw11** [reply]
> For me, this issue was caused because the IP for my vacuum changed which made the IP listed for the "host:" in the config.yaml incorrect. Once I updated that, everything loaded like normal. Also reserved that IP in my router to prevent that from happening going forward. Hope this helps!

I did the same, assigned the IP address for the roborock but still the same. I have to open the MiHome app and load the robot once to get the robot back to available in HA.

**Comment by adjbuzz** [comment]
It has still happened to me from time to time as well. Going to Developer
Tools and clicking to reload XIAOMI_CLOUD_MAP_EXTRACTOR also is a quick way
to get it to pull.

**Reply by lkw11** [reply]
> It has still happened to me from time to time as well. Going to Developer Tools and clicking to reload XIAOMI_CLOUD_MAP_EXTRACTOR also is a quick way to get it to pull.
> […](#)
> On Fri, Dec 9, 2022 at 10:18 PM lkw11 ***@***.***> wrote: For me, this issue was caused because the IP for my vacuum changed which made the IP listed for the "host:" in the config.yaml incorrect. Once I updated that, everything loaded like normal. Also reserved that IP in my router to prevent that from happening going forward. Hope this helps! I did the same, assigned the IP address for the roborock but still the same. I have to open the MiHome app and load the robot once to get the robot back to available in HA. — Reply to this email directly, view it on GitHub <[#298 (reply in thread)](https://github.com/PiotrMachowski/Home-Assistant-custom-components-Xiaomi-Cloud-Map-Extractor/discussions/298#discussioncomment-4360439)>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AHZEQFB5ZGIJVI75N5U5F3DWMQAA3ANCNFSM5XPONNGQ> . You are receiving this because you commented.Message ID: <PiotrMachowski/Home-Assistant-custom-components-Xiaomi-Cloud-Map-Extractor/repo-discussions/298/comments/4360439 @github.com>

I tried that and no luck for me. The only workaround for me at this stage is to open MiHome app and reload the roborock. 
have been research  and tried a lots different method but unfortunately situation remains the same

**Comment by chekangliang** [comment]
i also have the same issue I started observing this over the last month or so and the symptom (same as yours) happens intermittently - probably 50% of the week it is working. My device is roborock s8; connected through mi home app, running on the reserved IP. 

I try to read through a couple of debug thread and i find the logging on the cloud API does not provide much information. Does anyone know if the home assistant server call the cloud API, or a local API directly to the vacuum on a LAN. Maybe figure out where we can add logging to debug this better?

**Reply by PiotrMachowski** [reply]
Xiaomi Miio uses LAN access.
This integration connects to vacuum via LAN to retrieve map ID, then downloads a map from the cloud

**Reply by chekangliang** [reply]
It looks like it's a two step procedure : 

1) vacuum via LAN to retrieve map ID
2) then downloads a map from the cloud

is there any way to pull log to see how far the flow has gotten?

**Reply by PiotrMachowski** [reply]
You can enable debug logging
https://github.com/PiotrMachowski/Home-Assistant-custom-components-Xiaomi-Cloud-Map-Extractor#enabling-debug-logging

**Reply by chekangliang** [reply]
Follow the step and this is what came up: 
```

This error originated from a custom integration.

Logger: homeassistant.helpers.entity
Source: helpers/entity.py:548
First occurred: 12:42:15 PM (18 occurrences)
Last logged: 12:57:53 PM

Update of camera.xiaomi_cloud_map_extractor is taking over 10 seconds

Logger: custom_components.xiaomi_cloud_map_extractor.camera
Source: custom_components/xiaomi_cloud_map_extractor/camera.py:273
Integration: xiaomi_cloud_map_extractor (documentation, issues)
First occurred: 12:42:15 PM (2 occurrences)
Last logged: 12:51:26 PM

Got exception while fetching the state: Unable to discover the device 192.168.1.22
```

192.168.1.22 is the IP of my robovacuum (confirming it through the mihome app)
Does it mean it is failing on the LAN or on the cloud?

**Reply by PiotrMachowski** [reply]
It seems to be failing over LAN, you can try to restart your vacuum

**Comment by flepsman** [comment]
Hello & thanks for all the support you give us @PiotrMachowski 

I had the same issue recently, solved it using these forums & wanted to share my context if it helps anyone:

> After a power outage (dunno if related, i will update next time)
> I noticed that i had the "Failed to retrieve map from vacuum" error & blue screen on the card
> The Xiaomi Miio integration errors with the xiaomi devices added to HAOS
> Reloaded & fixed the Xiaomi Miio integration and devices but map was still with same error

The reason IMO was that when I had the power outage, the IP changed for my vacuum & most probably for all my xiomi devices.
Solved by using the token_extractor and adding the new IP to the secrets & restarting

During my attempts I shut down/started the vacuum once
Taking in account  that this does not write on the vacuum, the fact that i deleted one of the rooms in the card configuration was not the issue. Initially i was thinking this might be due to the fact that the maps got scrambled on the device.

I don't know if this: https://github.com/PiotrMachowski/Home-Assistant-custom-components-Xiaomi-Cloud-Map-Extractor/blob/master/blueprints/automation/disable_vacuum_camera_update_when_docked.yaml would help in any way the issue overall

I will update my comment if i think i have anything new.
Cheers!

**Reply by PiotrMachowski** [reply]
There are two different problems that can result in this error. The basic one is solvable by fixing IP/token. The more advanced one can potentially be fixed by the blueprint

**Reply by flepsman** [reply]
Hello @PiotrMachowski  thanks for your reply. I followed the discussion on and tried to add the blueprint but i fail to identify update_map_extractor.yaml to drop the lines. Should i create it? I don't have coding background so i might be missing something, sorry :(  (meantime i plan to look for some yaml course online to move from n00b level)

**Reply by PiotrMachowski** [reply]
@flepsman you don't need yaml at all to import blueprint:

[![Open your Home Assistant instance and show the blueprint import dialog with a specific blueprint pre-filled.](https://my.home-assistant.io/badges/blueprint_import.svg)](https://my.home-assistant.io/redirect/blueprint_import/?blueprint_url=https%3A%2F%2Fgithub.com%2FPiotrMachowski%2FHome-Assistant-custom-components-Xiaomi-Cloud-Map-Extractor%2Fblob%2Fmaster%2Fblueprints%2Fautomation%2Fupdate_map_extractor.yaml)

**Comment by dexonter** [comment]
This problem happens on and off - sometimes the map loads, and sometimes it does not...

The thing is...sometimes when the map does not load, it also fails to load in the Mi App...weirdly enough.

The vacuum has a reserved address, the internet connection last failed 417 days ago (I have a redundant connection anyway, 417 days ago one of them failed) - and it's very unpredictable and frustrating

Roborock S50 here

**Reply by PiotrMachowski** [reply]
Check out the discussion in this issue: https://github.com/PiotrMachowski/Home-Assistant-custom-components-Xiaomi-Cloud-Map-Extractor/issues/422

**Comment by eruisi** [comment]
I have the same issue:

![Screenshot 2023-09-06 at 1 48 06 PM](https://github.com/PiotrMachowski/Home-Assistant-custom-components-Xiaomi-Cloud-Map-Extractor/assets/26517770/66359718-d65e-4180-9ebf-60849ff40d0e)

I don't think it is credential issue because I can control the vacumm with the buttons on the map card. Just cannot get the map from it.

How to debug and fix the issue?

**Reply by PiotrMachowski** [reply]
https://github.com/PiotrMachowski/Home-Assistant-custom-components-Xiaomi-Cloud-Map-Extractor/issues/422
